android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotification.java

```diff
@@ -78,6 +78,9 @@ private Bundle getBundleFromIntent(Intent intent) {
             bundle = intent.getBundleExtra("notification");
         } else if (intent.hasExtra("google.message_id") || intent.hasExtra("lp_messageId")) {
             bundle = intent.getExtras();
+        } else if (intent.getData() != null) {
+            intent.putExtra("uri", intent.getData());
```

Comment:
instead of modifying intent, I would just modify the bundle, just as a safety measure in case modifying the intent causes a side effect
```
bundle = intent.getExtras();
bundle.putParcelable("uri", intent.getData())
```
something like this, with the proper typing if I didn't get it right